StoryBook 페이지 개발

metacode22 opened the pull request:
#22 

![image](https://user-images.githubusercontent.com/93233930/211500495-bfb12de2-fa03-4bc0-b081-13f7a8cd47c2.png)
## 작업 목록
- 연도별로 story들이 묶이도록 구현.
  - 위에서 내림차순으로, 해당 연도에서 오른쪽으로는 오름차순으로 정렬됨.
- image 테스트를 위해 다음 계정으로 더미 데이터 생성
```
신승준 계정
email: alohajune22@gmail.com
password: metamong,
fullName: Shin Seung-Jun
_id: 63bcf0d4f596c65f9ee2f226
token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7Il9pZCI6IjYzYmNmMGQ0ZjU5NmM2NWY5ZWUyZjIyNiIsImVtYWlsIjoiYWxvaGFqdW5lMjJAZ21haWwuY29tIn0sImlhdCI6MTY3MzMyNjgwNH0.JGg2IJjc-iBZF9Yxouln74cG3FqDXmMcUgXFjafFSyE
```

## 예정
- realTitle 등 구체적인 속성명이나 값은 유리님과 논의 후 맞춰서 변경할 예정.
- api는 추후 따로 빼서 사용할 예정.
- 스타일 추후 변경 예정.

## 궁금한 점
- image가 없는 story일 경우 defaultImage로 어떤 걸 이용할까요? 나중에 스타일 좀 정해지면 그 때 정할까요? 지금은 그냥 아무거나 가져왔습니다.😀
- 코드량이 많고 커밋을 잘 못 끊었는데 다음 부터는 잘 나눌게요!

metacode22 committed:
feature: useFetchStories 커스텀 훅 구현

- api 호출은 추후 모듈화 예정

metacode22 committed:
feature: StoryBook 페이지 개발

- 연도별로 story들이 묶이도록 구현
- StoryCard image의 threshold 0.5에서 0.3으로 변경
- error message 상수
- story, title interface 분리

metacode22 committed:
fix: userId에 따라 storybook을 불러오도록 변경

metacode22 committed:
remove: StoryBookDummyData.tsx 파일 삭제

metacode22 committed:
remove: StoryBookDummyData.tsx 파일 삭제

tooooo1 commented on src/hooks/useFetchStories.ts:
```diff
@@ -0,0 +1,85 @@
+          year: yearA,
+          month: monthA,
+          day: dayA,
+        }: Pick<Title, 'year' | 'month' | 'day'> = JSON.parse(storyA.title);
```
Pick이 너무 많이 쓰이는 것 같아요. 유형도 똑같구요.
Title을 좀 더 세분화해서 extends로 확장하면 유틸리티 함수를 사용하지 않아도 될 것 같아요.

metacode22 replied:
넵 덕분에 Pick이 줄어들었네요!

tooooo1 reviewed (approved):
고생하셨어요! 타입 부분은 한번 고려해 보시는 건 어떨까요?

metacode22 commented on src/pages/StoryBook.tsx:
```diff
@@ -0,0 +1,39 @@
+
+        return (
+          <StoriesByYear
+            key={year as string}
```
오 맞네요! 구웃 없애도 에러 안뜨네요. 반영하겠습니다.

metacode22 replied:
- 기존 [[2022, story, story...], [1995, story, story...]]로 렌더링하는 방식에서
  - [{year: 2022, stories: [story, story...]}, {year: 1995, stories: [story, story...]}]로 변경

metacode22 committed:
feature: StoryBook 렌더링 방식 변경

- 기존 [[2022, story, story...], [1995, story, story...]]로 렌더링하는 방식에서
  - [{year: 2022, stories: [story, story...]}, {year: 1995, stories: [story, story...]}]로 변경
- interface 확장성 개선
- errorMessage -> errorMessages로 파일 이름 변경

metacode22 committed:
fix: resolve conflict

metacode22 committed:
refactor: useParams를 통해 특정 사용자의 stories를 받아오도록 변경

tooooo1 commented on src/interfaces/story.ts:
```diff
@@ -0,0 +1,22 @@
+  image: string;
+}
+
+export interface StoryDate {
+  year: string;
+  month: string;
+  day: string;
+}
+
+export interface Title {
+  realTitle: string;
+}
+
+export interface StoriesWithYear {
+  year: string;
+  stories: Story[];
+}
```
year가 중복되었네요. Pick으로도 여러 번 쓰이구요!
이제 year 타입을 만들어서 조합하는 것도 나쁘지 않을 거 같아요

tooooo1 commented on src/hooks/useFetchStories.ts:
```diff
@@ -0,0 +1,87 @@
+    const storiesWithYear: StoriesWithYear[] = [];
+
+    stories.forEach((story: Story) => {
+      const { year }: Pick<StoryDate, 'year'> = JSON.parse(story.title);
+      yearsSet.add(year);
+    });
+
+    yearsSet.forEach((year) => {
+      const storiesFilteredByYear: Story[] = stories.filter((story: Story) => {
+        const { year: yearOfStory }: Pick<StoryDate, 'year'> = JSON.parse(
```
그러면 해당 PR에서도 위 두개 Pick 지울 수 있을 거 같아요.

tooooo1 reviewed (changes requested):
고생하셨습니다앙 몇가지 코멘트 남길게요

metacode22 committed:
refactor: story interface 확장성 개선

tooooo1 reviewed (approved):
좋아요
